yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -271,6 +271,33 @@ private[spark] class Client(
         appContext.setResource(capability)
     }
 
+    sparkConf.get(ROLLED_LOG_INCLUDE_PATTERN).foreach { includePattern =>
```

Comment:
Yes, I think your understanding is also right. In my previous assumption, exclude patterns will only be effective when include pattern is set, this might be OK. But I think your change should also be fine.